### PR TITLE
Fix web3-eth-abi export types

### PR DESCRIPTION
**Pull request by Velenir**
## Description

Currently if I want to use `web3-eth-abi` separate from the whole `web3` in Typescript I get wrong types on import
Declarations in `index.d.ts` specify `export class AbiCoder` as the only export, so TS expects the following to work
```ts
import { AbiCoder } from 'web3-eth-abi'
const  coder = new AbiCoder()
coder.encodeFunctionSignature('myMethod(uint256,string)')
```

But in reality `src/index.js` exports a singleton instance of `AbiCoder` as default
```js
var coder = new ABICoder();

module.exports = coder;
```

And TS code breaks with `Uncaught TypeError: web3_eth_abi__WEBPACK_IMPORTED_MODULE_1__.AbiCoder is not a constructor`
To properly use abi I have to 
```ts
import Web3Abi, { AbiCoder } from 'web3-eth-abi'

const Abi = (Web3Abi as unknown) as AbiCoder
```

To sum it up, TS declarations export named export class, whereas real JS export is default of that class instance.
I think there's a similar problem (sans singleton aspect) in `web3-eth-ens` and  `web3-eth-contract`, maybe more. Didn't look too deep into it.

This PR switches named export of `class AbiCoder` to type only export (as JS doesn't export class at all), and adds a default export of `AbiCoder` instance.
As an alternative, you may want to export both class and singleton instance in both JS and TS.

I believe #3198 is a related issue.

## Type of change

Type fix

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build-all` and tested the resulting files from `dist` folder in a browser.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.

### CODE REVIEW

**Comment by cgewecke** [comment]
@Velenir Apologies for the delay reviewing here...

It looks like this issue affects ~16 Web3 sub-modules and various PR's have been opened in piecemeal fashion to address each one as users have run into the bug.

One complication with fixing it is that the solution should (ideally) be compatible with near-term plans to rewrite Web3 1.x in TS. Because the types are wrong for constructor usage but people are already using them as definitions, doing this correctly may require a breaking change. 

TLDR; I think we need to fix this in a consistent way across the project and have a plan for how to manage its implications.

Related issues include
+ #3215 
+ #3292 
+ #3198 

cc: @GregTheGreek

**Comment by GregTheGreek** [comment]
FYI We've opened up #3606 to try and track these changes in a more consistent manner. this PR isn't necessarily off the table, I just think we need to asses everything first

**Comment by DrZainAzm** [comment]
“Tools”,“Interact”,”Contracts”
Contract Address: https://etherscan.io/address/0x0000000000000000000000000000000000000000
ABI: http://api.etherscan.io/api?module=contract&action=getabi&address=0x0000000000000000000000000000000000000000&format=raw
Spender address (migration contract): 0x3c75d37b579e0e4896f02c0122baa4de05383a6a
 “value:uint256” 8412000000000000000000
“Write”,“Generate transaction”,“Send transaction”

https://etherscan.io/token/0xed5a231ecc0fa775980ad7b86e49872feacd4c8f?a=0x3c5459bcde2d5c1edc4cc6c6547d6cb360ce5ae9



From":"0x0000000000000000000000000000000000000000","contractAddress":"0xed5a231ecc0fa775980ad7b86e49872feacd4c8f","to":"0xc287cf1f50d5bd56de70bdec1f4103d2b733a6d4","value":"8412000000000000000000","tokenName":"LuckyBucks","tokenSymbol":"LBT","tokenDecimal":"18"


Malaysia needs a privacy law that protects people, not corporations.

@Privacy rights  act 1974 and Securities Law act 1933 are violated by the Lucky Bucks, LLC., 5820 Live Oak Pkwy Ste 300, Norcross, 30071 Georgia, United States,  Phone: +1 (770) 449-4699, E-mail: Inquiries@luckybucksga.com, Website: https://luckybucksga.com (Subsidiary).

Head Office address: Seven Aces Limited, 79 Wellington Street West, Suite 1630, PO Box 138, Toronto, Ontario, Canada M5K 1H1, Phone: +1 416.569-3292, Fax: +1 416.477-3401, Website: http://www.sevenaces.com, E-mail: Info@sevenaces.com, Email: Manish@sevenaces.com

**Review by jdevcs — COMMENTED** [review]
Closing this PR as there is no activity from long time. Recently merged similar fix PR https://github.com/ChainSafe/web3.js/pull/4937.
